external/avro/src/test/scala/org/apache/spark/sql/avro/AvroCatalystDataConversionSuite.scala

```diff
@@ -246,4 +270,34 @@ class AvroCatalystDataConversionSuite extends SparkFunSuite
     }.getMessage
     assert(message ==  "Cannot convert Catalyst type StringType to Avro type \"long\".")
   }
+
+  test("deserialize Record with generic list") {
+    val jsonFormatSchema =
+      """
+        |{ "type": "record",
+        |  "name": "record",
+        |  "fields" : [{
+        |    "name": "array",
+        |    "type": {
+        |      "type": "array",
+        |      "items": "int"
+        |    }
+        |  }]
+        |}
+      """.stripMargin
+    val avroSchema = new Schema.Parser().parse(jsonFormatSchema)
+    val data = new GenericRecordBuilder(avroSchema)
+      .set("array", Collections.emptyList())
```

Comment:
How about using an array with elements (e.g. `[1, 2]`) and read it back with result validation?

Reply:
Added this check, also made the field optional.